src/repo.rs

```diff
@@ -1712,6 +1713,71 @@ impl Repository {
         }
     }
 
+    /// Save the local modifications to a new stash.
+    pub fn stash_save(&mut self,
+                      stasher: &Signature,
+                      message: &str,
+                      flags: Option<StashFlags>)
+                      -> Result<Oid, Error> {
+        unsafe {
+            let mut raw_oid = raw::git_oid { id: [0; raw::GIT_OID_RAWSZ] };
+            let message = try!(CString::new(message));
+            let flags = flags.unwrap_or(StashFlags::empty());
+            try_call!(raw::git_stash_save(&mut raw_oid,
+                                          self.raw(),
+                                          stasher.raw(),
+                                          message,
+                                          flags.bits() as c_uint));
+            Ok(Binding::from_raw(&raw_oid as *const _))
+        }
+    }
+
+    /// Apply a single stashed state from the stash list.
+    pub fn stash_apply(&mut self,
+                       index: usize,
+                       opts: Option<&mut StashApplyOptions>)
+                       -> Result<(), Error> {
+        unsafe {
+            let opts = opts.map(|opts| opts.raw());
+            try_call!(raw::git_stash_apply(self.raw(), index, opts));
+            Ok(())
+        }
+    }
+
+    /// Loop over all the stashed states and issue a callback for each one.
+    ///
+    /// Return `true` to continue iterating or `false` to stop.
+    pub fn stash_foreach<C>(&mut self, mut callback: C) -> Result<(), Error>
+        where C: FnMut(usize, &str, &Oid) -> bool
+    {
+        unsafe {
+            let mut data = Box::new(StashCbData { callback: &mut callback });
```

Comment:
I believe the `Box` can be elided here and you can just pass a stack pointer, right?